libs/oeth/swap/src/state.ts

```diff
@@ -0,0 +1,33 @@
+import { useState } from 'react';
+
+import { tokens } from '@origin/shared/contracts';
+import { useDebouncedEffect } from '@react-hookz/web';
+import { produce } from 'immer';
+import { createContainer } from 'react-tracked';
+
+import type { Token } from '@origin/shared/contracts';
+
+export const { Provider: SwapProvider, useTracked: useSwapState } =
+  createContainer(() => {
```

Comment:
This is interesting, never used `react-tracked` but I like the interface

Reply:
very cool library made by the creator of Zustand, Jotai and Valtio. Super clean and lean api, just a small wrapper around context api that provides proxy support. It allows for granular re-renders of children of context providers and this kind of pattern for "subscribing" to state changes.